### PR TITLE
JAMES-3431 Mock SMTP server ReceivedMailRepository keep ESMTP parameters as part of the envelope

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

**Comment by chibenwa** [comment]
`docker: Error response from daemon: No such container: git-publish-dsn-mock-smtp-45c7d5db-d180-42b9-9147-6d5521a48cc4.`

test this please

File: server/mailet/mock-smtp-server/src/test/java/org/apache/james/mock/smtp/server/HTTPConfigurationServerTest.java [hunk 1]
```diff
@@ -177,7 +177,7 @@ void getShouldReturnPreviouslyStoredData() {
             mailRepository.store(MailsFixutre.MAIL_1);
 
             String response = when()
-                    .get()
+                    .get().prettyPeek()
```
Comment (Arsnael) [inline server/mailet/mock-smtp-server/src/test/java/org/apache/james/mock/smtp/server/HTTPConfigurationServerTest.java:190]: debug spotted

File: server/mailet/mock-smtp-server/src/test/java/org/apache/james/mock/smtp/server/HTTPConfigurationServerTest.java [hunk 2]
```diff
@@ -194,15 +194,43 @@ void getShouldReturnMultipleEmails() {
             mailRepository.store(MailsFixutre.MAIL_2);
 
             String response = when()
-                    .get()
+                    .get().prettyPeek()
```
Comment (Arsnael) [inline server/mailet/mock-smtp-server/src/test/java/org/apache/james/mock/smtp/server/HTTPConfigurationServerTest.java:207]: debug spotted

File: server/mailet/mock-smtp-server/src/test/java/org/apache/james/mock/smtp/server/HTTPConfigurationServerTest.java [hunk 3]
```diff
@@ -78,6 +79,15 @@ void getShouldReturnEmptyByDefault() {
                 .body(".", hasSize(0));
         }
 
+        @Test
+        void shouldExposeVersion() {
+            given()
+                .basePath("/version")
+                .get().prettyPeek()
```
Comment (Arsnael) [inline server/mailet/mock-smtp-server/src/test/java/org/apache/james/mock/smtp/server/HTTPConfigurationServerTest.java:86]: debug spotted

File: server/mailet/mock-smtp-server/src/main/java/org/apache/james/mock/smtp/server/model/Mail.java [hunk 4]
```diff
@@ -20,61 +20,251 @@
...
+        public static Builder builder() {
+            return new Builder();
+        }
+
+        private final String name;
+        private final String value;
+
+        private Parameter(String name, String value) {
+            Preconditions.checkNotNull(name, "'name' field cannot be null");
+            Preconditions.checkNotNull(value, "'value' field cannot be null");
```
Comment (LanKhuat) [inline server/mailet/mock-smtp-server/src/main/java/org/apache/james/mock/smtp/server/model/Mail.java:78]: is double checking the fields necessary?

**Review by Arsnael — APPROVED** [review]

**Review by LanKhuat — APPROVED** [review]

**Comment by Arsnael** [comment]
Was green before CI restart

But needs rebase

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
test this please

**Comment by Arsnael** [comment]
Merged

**Pull request closed without merging**